crates/mev-share-rpc-api/src/types.rs

```diff
@@ -109,4 +316,97 @@ mod tests {
         let res: Result<Vec<SendBundleRequest>, _> = serde_json::from_str(str);
         assert!(res.is_ok());
     }
+
+    #[test]
+    fn can_deserialize_complex() {
+        let str = r#"
+        [{
+            "version": "v0.1",
+            "inclusion": {
+                "block": "0x1"
+            },
+            "body": [{
+                "tx": "0x02f86b0180843b9aca00852ecc889a0082520894c87037874aed04e51c29f582394217a0a2b89d808080c080a0a463985c616dd8ee17d7ef9112af4e6e06a27b071525b42182fe7b0b5c8b4925a00af5ca177ffef2ff28449292505d41be578bebb77110dfc09361d2fb56998260",
+                "canRevert": false
+            }], 
+            "privacy": {
+                "hints": [
+                  "calldata"
+                ]
```

Comment:
does this work as vec?
I suspect we need a custom serialize/deserialize impl for `PrivacyHint` that supports vec, but perhaps this works out of the box

Reply:
sorry, could you clarify? Not sure I understand what you're saying needs to change in the custom serialize/deserialize impl for `PrivacyHint` that we have in the current PR

Reply:
hints is a vec which we're trying to serialize into an object, I'm not sure this works out of the box with serde_json, but maybe it does?

I was suggesting some additional serde roundtrip tests for this, it could be that this just works, the existing test at least makes it look like it does.

Reply:
added more tests 👍